backend/danswer/connectors/document360/connector.py

```diff
@@ -0,0 +1,130 @@
+import requests
+from datetime import datetime, timezone
+from typing import Any, Generator, Optional, List, Union
+from danswer.connectors.interfaces import LoadConnector, PollConnector, GenerateDocumentsOutput, SecondsSinceUnixEpoch
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.models import Document, ConnectorMissingCredentialError
+
+DOCUMENT360_BASE_URL = "https://preview.portal.document360.io/"
+DOCUMENT360_API_BASE_URL = "https://apihub.document360.io/v2"
+
+class Document360Connector(LoadConnector, PollConnector):
+    def __init__(
+        self,
+        workspace: str,
+        categories: List[str],
+        portal_id: Optional[str] = None,
+        api_token: Optional[str] = None
+    ) -> None:
+        self.portal_id = portal_id
+        self.workspace = workspace
+        self.categories = categories
+        self.api_token = api_token
+
+    def load_credentials(self, credentials: dict[str, Any]) -> Optional[dict[str, Any]]:
+        self.api_token = credentials.get("document360_api_token")
+        self.portal_id = credentials.get("portal_id")
+        return None
+
+    def _make_request(self, endpoint: str, params: Optional[dict] = None) -> Any:
+        if not self.api_token:
+            raise ConnectorMissingCredentialError("Document360")
+
+        headers = {
+            'accept': "application/json",
+            'api_token': self.api_token
+        }
+
+        response = requests.get(f"{DOCUMENT360_API_BASE_URL}/{endpoint}", headers=headers, params=params)
+        response.raise_for_status()
+
+        return response.json()['data']
+
+    def _get_workspace_id_by_name(self):
+        projects = self._make_request("ProjectVersions")
+        return next((project['id'] for project in projects if project["version_code_name"] == self.workspace), None)
```

Comment:
Is the user provides an invalid workspace, the connector should fail to alert the user that something has gone wrong. The error message for the connector is passed to the frontend so we should add a informative message

Reply:
Okay, I will handle this error.